src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -67,22 +67,25 @@ public IEnumerable<SarifLog> Match(IEnumerable<SarifLog> previousLogs, IEnumerab
                 {
                     baselineRuns = runsByToolPrevious[key];
                 }
-                IEnumerable<Run> currentRuns = new Run[0];
 
+                IEnumerable<Run> currentRuns = new Run[0];
                 if (runsByToolCurrent.ContainsKey(key))
                 {
                     currentRuns = runsByToolCurrent[key];
                 }
 
-                resultToolLogs.Add(BaselineSarifLogs(baselineRuns, currentRuns));
+                if (currentRuns.Any())
+                {
+                    resultToolLogs.Add(BaselineSarifLogs(baselineRuns, currentRuns));
+                }
             }
 
             return new List<SarifLog> { resultToolLogs.Merge(mergeEmptyLogs: true) };
         }
 
         private static Dictionary<string, List<Run>> GetRunsByTool(IEnumerable<SarifLog> sarifLogs)
         {
-            Dictionary<string, List<Run>> runsByTool = new Dictionary<string, List<Run>>();
+            var runsByTool = new Dictionary<string, List<Run>>();
```

Comment:
>Dictionary [](http://example.com/codeflow?start=33&length=10)

Call the IEqualityComparer ctor here and pass StringComparer.OrdinalIgnoreCase to get case-insensitive key comparisons using .ContainsKey

Reply:
done